Packet++/src/BgpLayer.cpp

```diff
@@ -744,9 +745,25 @@ namespace pcpp
 
 		if (newNlriDataLen > curNlriDataLen)
 		{
+
+			// offsetInLayer, numOfBytesToExtend
+			//		int indexToInsertData = layer->m_Data + offsetInLayer - m_RawPacket->getRawData();
+			auto bytesToExtend = newNlriDataLen - curNlriDataLen;
+
+			if (m_Data != nullptr && m_Packet != nullptr)
+			{
+				auto raw_len = static_cast<size_t>(m_Packet->getRawPacket()->getRawDataLen());
+				if (raw_len + bytesToExtend < raw_len)
```

Comment:
How can this be true? 🤔 
`bytesToExtend` is of type `size_t`, which can't be a negative number

Reply:
As previously discussed, this can overflow when unsigned.

Reply:
You're right! Can you add a comment above this line to remind future us why we did it? 🙂